java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,376 @@
+    if (methodSelect.is(Tree.Kind.IDENTIFIER)) {
+      return true;
+    } else if (methodSelect.is(Tree.Kind.MEMBER_SELECT)) {
+      MemberSelectExpressionTree memberSelectExpression = (MemberSelectExpressionTree) methodSelect;
+      ExpressionTree target = memberSelectExpression.expression();
+      if (target.is(Tree.Kind.IDENTIFIER)) {
+        IdentifierTree identifier = (IdentifierTree) target;
+        return THIS_SUPER.contains(identifier.name());
+      }
+    }
+    return false;
+  }
+
+  private void resetConstraintsOnFields() {
+    boolean changed = false;
+    Map<Symbol, SymbolicValue> values = Maps.newHashMap(programState.values);
+    for (Entry<Symbol, SymbolicValue> entry : values.entrySet()) {
+      if (constraintManager.isConstrained(programState, entry.getValue())) {
+        Symbol symbol = entry.getKey();
+        if (isField(symbol)) {
+          VariableTree variable = ((Symbol.VariableSymbol) symbol).declaration();
+          if (variable != null) {
+            changed = true;
+            SymbolicValue nonNullValue = constraintManager.supersedeSymbolicValue(variable);
+            values.put(symbol, nonNullValue);
+          }
+        }
+      }
+    }
+    if (changed) {
+      programState = new ProgramState(values, programState.constraints);
+    }
+  }
+
+  private static boolean isField(Symbol symbol) {
+    return symbol.isVariableSymbol() && !symbol.owner().isMethodSymbol();
+  }
+
+  private void setSymbolicValueNullValue(VariableTree variableTree) {
+    programState = put(programState, variableTree.symbol(), SymbolicValue.NULL_LITERAL);
+  }
+
+  private void setSymbolicValueForEachValue(VariableTree variableTree) {
+    SymbolicValue val = constraintManager.createSymbolicValue(variableTree);
+    programState = put(programState, variableTree.symbol(), val);
+  }
+
+  private void setSymbolicValueFalseValue(VariableTree variableTree) {
+    programState = put(programState, variableTree.symbol(), SymbolicValue.FALSE_LITERAL);
+  }
+
+  private void setSymbolicValueValueFromInitializer(VariableTree variableTree, ExpressionTree initializer) {
+    SymbolicValue val = constraintManager.eval(programState, initializer);
+    programState = put(programState, variableTree.symbol(), val);
+  }
+
+  static ProgramState put(ProgramState programState, Symbol symbol, SymbolicValue value) {
+    SymbolicValue symbolicValue = programState.values.get(symbol);
+    // update program state only for a different symbolic value
+    if (symbolicValue == null || !symbolicValue.equals(value)) {
+      Map<Symbol, SymbolicValue> temp = Maps.newHashMap(programState.values);
+      temp.put(symbol, value);
+      return new ProgramState(temp, programState.constraints);
+    }
+    return programState;
+  }
+
+  public void enqueue(ExplodedGraph.ProgramPoint programPoint, ProgramState programState) {
+    ExplodedGraph.Node node = explodedGraph.getNode(programPoint, programState);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "node" which hides the field declared at line 72. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)